packages/monaco-editor-wrapper/src/editorVscodeApi.ts

```diff
@@ -1,34 +1,47 @@
-import { MonacoEditorBase } from './editor.js';
+import { EditorAppBase, EditorAppConfig, EditorAppType, VscodeUserConfiguration } from './editor.js';
 import { updateUserConfiguration } from 'vscode/service-override/configuration';
 import { registerExtension, IExtensionManifest, ExtensionHostKind } from 'vscode/extensions';
 import 'vscode/default-extensions/theme-defaults';
-import { MonacoEditorWrapper } from './wrapper.js';
+import { UserConfig } from './wrapper.js';
 
-export type VscodeUserConfiguration = {
-    json?: string;
-}
-
-export type EditorVscodeApiConfig = {
+export type EditorAppConfigVscodeApi = EditorAppConfig & {
+    editorAppType: 'vscodeApi';
     extension?: IExtensionManifest | object;
     extensionFilesOrContents?: Map<string, string | URL>;
     userConfiguration?: VscodeUserConfiguration;
-}
+};
 
-export class EditorVscodeApi extends MonacoEditorBase implements MonacoEditorWrapper {
+export class EditorAppVscodeApi extends EditorAppBase {
+
+    constructor(id: string, userConfig: UserConfig) {
+        super(id, userConfig);
+        const userInput = userConfig.wrapperConfig.editorAppConfig as EditorAppConfigVscodeApi;
+        this.getAppConfig().userConfiguration = userInput.userConfiguration ?? undefined;
+        this.getAppConfig().extension = userInput.extension ?? undefined;
+        this.getAppConfig().extensionFilesOrContents = userInput.extensionFilesOrContents ?? undefined;
+    }
+
+    getAppType(): EditorAppType {
+        return 'vscodeApi';
+    }
+
+    getAppConfig(): EditorAppConfigVscodeApi {
+        return this.appConfig as EditorAppConfigVscodeApi;
+    }
 
     async init() {
-        const wrapperConfig = this.monacoConfig as EditorVscodeApiConfig;
-        if (wrapperConfig.extension) {
-            const extension = wrapperConfig.extension as IExtensionManifest;
+        if (this.getAppConfig().extension) {
```

Comment:
If we need to get the config repeatedly, we could just use `const config = this.getAppConfig()`, and then save us the repeated method calls later on here.

Reply:
Done ⬇️